Rdev/xgx_stat_ci_ordinal.R

```diff
@@ -0,0 +1,335 @@
+#' @param fun.args Optional additional arguments passed on to the functions.
+#' @param na.rm If FALSE, the default, missing values are removed with a 
+#' warning. If TRUE, missing values are silently removed.
+#' @param show.legend logical. Should this layer be included in the legends? 
+#' NA, the default, includes if any aesthetics are mapped. FALSE never 
+#' includes, and TRUE always includes.
+#' @param inherit.aes If FALSE, overrides the default aesthetics, rather 
+#' than combining with them. This is most useful for helper functions that 
+#' define both data and aesthetics and shouldn't inherit behaviour from the 
+#' default plot specification, e.g. borders.
+#' @param ... other arguments passed on to layer. These are often aesthetics, 
+#' used to set an aesthetic to a fixed value, like color = "red" or size = 3. 
+#' They may also be parameters to the paired geom/stat.
+#'
+#' @return ggplot2 plot layer
+#'
+#' @examples
+#' # default settings for normally distributed data, 95% confidence interval,  
+#' data <- data.frame(x = rep(c(1, 2, 3), each = 20),
+#'                    y = rep(c(1, 2, 3), each = 20) + stats::rnorm(60),
+#'                    group = rep(1:3, 20))
+#' xgx_plot(data, ggplot2::aes(x = x, y = y)) + 
+#'   xgx_summary_ci(conf_level = 0.95)
+#' 
+#' # try different geom 
+#' xgx_plot(data, ggplot2::aes(x = x, y = y)) + 
+#'   xgx_summary_ci(conf_level = 0.95, geom = list("ribbon", "point", "line"))
+#'  
+#' # plotting lognormally distributed data
+#' data <- data.frame(x = rep(c(1, 2, 3), each = 20),
+#'                    y = 10^(rep(c(1, 2, 3), each = 20) + stats::rnorm(60)),
+#'                    group = rep(1:3, 20))
+#' xgx_plot(data, ggplot2::aes(x = x, y = y)) + 
+#'   xgx_summary_ci(conf_level = 0.95, distribution = "lognormal")
+#'   
+#' # note: you DO NOT need to use both distribution = "lognormal"
+#' # and scale_y_log10()
+#' xgx_plot(data, ggplot2::aes(x = x, y = y)) + 
+#'   xgx_summary_ci(conf_level = 0.95) + xgx_scale_y_log10()
+#'  
+#' # plotting binomial data
+#' data <- data.frame(x = rep(c(1, 2, 3), each = 20),
+#'                    y = stats::rbinom(60, 1, rep(c(0.2, 0.6, 0.8),
+#'                    each = 20)),
+#'                    group = rep(1:3, 20))
+#' xgx_plot(data, ggplot2::aes(x = x, y = y)) + 
+#'   xgx_summary_ci(conf_level = 0.95, distribution = "binomial")
+#'  
+#' # including multiple groups in same plot
+#' xgx_plot(data, ggplot2::aes(x = x, y = y)) + 
+#'   xgx_summary_ci(conf_level = 0.95, distribution = "binomial", 
+#'               ggplot2::aes(color = factor(group)),
+#'               position = ggplot2::position_dodge(width = 0.5))
+#'  
+#' @importFrom stats rnorm
+#' @importFrom stats rbinom
+#' @importFrom stats na.omit
+#' @importFrom stats qt
+#' @importFrom stats var
+#' @importFrom binom binom.exact
+#' @importFrom ggplot2 stat_summary
+#' @importFrom ggplot2 aes
+#' @importFrom ggplot2 position_dodeg
+#' @export
+xgx_stat_ci <- function(
```

Comment:
wait, you're defining xgx_stat_ci again?  will this overwrite the definition in xgx_stat_ci.R?